File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {

        map<int,int> mp;
        for(int i=0;i<nums.size();i++)
            mp[nums[i]]++;


        vector<pair<int,int>> v;

        for(auto it: mp)
            v.push_back({it.second,it.first});

        sort(v.begin(),v.end());


        vector<int> res;

        int n=v.size()-1;

        while(res.size()<k && n>=0){
            res.push_back(v[n].second);
            n--;
        }

        return res;        
    }
};

File: Data Structures & Algorithms/jump-game-ii/submission-0.java
class Solution {
    public int jump(int[] nums) {
        int n = nums.length;
        int []dp = new int[n];

        for(int i=0;i<n;i++){
            dp[i]=n+1;
        }
        int jump;
        dp[0]=0;
        for(int i=0;i<nums.length;i++){
            jump = nums[i];
            for(int j=i+1;j<=Math.min(n-1,i+jump);j++)
                dp[j]=Math.min(dp[j],dp[i]+1);
        }
        return dp[n-1];
    }
}
